aggregate_root/spec/instrumented_repostory_spec.rb

```diff
@@ -0,0 +1,86 @@
+require 'spec_helper'
+require 'active_support/notifications'
+
+module AggregateRoot
+  RSpec.describe InstrumentedRepository do
+
+    describe "#load" do
+      specify "wraps around original implementation" do
+        some_repository = spy
```

Comment:
Is there a way for spy to check if it still follows an API that is mocked? I'm always wary of such tests. I briefly remember https://github.com/psyho/bogus implementing this check.

Reply:
https://relishapp.com/rspec/rspec-mocks/v/3-8/docs/verifying-doubles/using-an-instance-double